### PR TITLE
GSoC-2020- writing tests and clean code [Week 9]

### PR DESCRIPTION
**Pull request by prakashupes**

### CODE REVIEW

**Comment by cvvergara** [comment]
What is this file for?
`.vscode/settings.json`

File: include/bipartite/pgr_bipartite_driver.hpp [hunk 1]
```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
-            V_i v, vend;
-
-         // iterate through every vertex in the graph
+                std::vector <boost::default_color_type> partition (graph.num_vertices());    
+                auto partition_map =
+                    make_iterator_property_map(partition.begin (), boost::get (boost::vertex_index, graph.graph));
+
+               /* abort in case of an interruption occurs (e.g. the query is being cancelled) */
+                CHECK_FOR_INTERRUPTS();
+                boost::is_bipartite (graph.graph, boost::get (boost::vertex_index, graph.graph), partition_map);
```
Comment (cvvergara) [inline include/bipartite/pgr_bipartite_driver.hpp:65]: You removed the try/catch please keep it

File: include/bipartite/pgr_bipartite_driver.hpp [hunk 2]
```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
-                     int64_t vid = graph[*v].id;
-                     auto color =(boost::get (partition_map, *v) == boost::color_traits <boost::default_color_type>::white () ? "white" : "black");
-                     if(color=="white")
-                     {
-                         results.push_back(
-                             {
+                    int64_t vid = graph[*v].id;
+                    auto color =(boost::get (partition_map, *v) == boost::color_traits <boost::default_color_type>::white () ? "white" : "black");
+                    if(color=="white"){
+                        results.push_back({
```
Comment (cvvergara) [inline include/bipartite/pgr_bipartite_driver.hpp:72]: Put in one line

File: include/bipartite/pgr_bipartite_driver.hpp [hunk 3]
```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
-                         );
-                     }
-                     else
-                     {
-                         results.push_back(
-                             {
+                             });
+                    }
+                    else{
+                        results.push_back({
```
Comment (cvvergara) [inline include/bipartite/pgr_bipartite_driver.hpp:77]: Put in one line

File: include/bipartite/pgr_bipartite_driver.hpp [hunk 4]
```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
-
-                     int64_t vid = graph[*v].id;
-                     auto color =(boost::get (partition_map, *v) == boost::color_traits <boost::default_color_type>::white () ? "white" : "black");
-                     if(color=="white")
-                     {
-                         results.push_back(
-                             {
+                    int64_t vid = graph[*v].id;
+                    auto color =(boost::get (partition_map, *v) == boost::color_traits <boost::default_color_type>::white () ? "white" : "black");
+                    if(color=="white"){
```
Comment (cvvergara) [inline include/bipartite/pgr_bipartite_driver.hpp:71]: This whole if can be done in one line:
```
(color=="white")? results.push_back({vid,0}) : results.push_back({vid,1})
```

File: include/bipartite/pgr_bipartite_driver.hpp [hunk 5]
```diff
@@ -41,96 +41,63 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
-
-
-
-
+                    G &graph ){
+
+                std::vector<pgr_bipartite_rt> results(graph.num_vertices());
+                  
+                bool bipartite = boost::is_bipartite (graph.graph);
+                if(bipartite)
```
Comment (cvvergara) [inline include/bipartite/pgr_bipartite_driver.hpp:93]: In one line
```
                if(bipartite) results=print_Bipartite(graph);
```

File: run.sh [hunk 6]
```diff
@@ -1,8 +1,150 @@
+#!/bin/bash
```
Comment (cvvergara) [inline run.sh:1]: This file should not be included on the repo, please delete it

**Comment by cvvergara** [comment]
Issues with documentation:

https://github.com/pgRouting/GSoC-pgRouting/runs/912238094#step:7:224
https://github.com/pgRouting/GSoC-pgRouting/runs/912238094#step:7:1051

I dont understand why functions that you didnt do mark an error

**Comment by cvvergara** [comment]
What happened? why Prim & Kruskal are messed up?
https://github.com/pgRouting/GSoC-pgRouting/pull/112/checks?check_run_id=912238179#step:7:225

File: src/LTDTree/LTDTree.c [hunk 7]
```diff
@@ -27,29 +27,19 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
...
 #include "c_common/time_msg.h"
 #include "c_types/pgr_ltdtree_rt.h"
 #include "c_common/edges_input.h"
-#include "c_common/arrays_input.h"
+
 #include "drivers/LTDTree/LTDTree_driver.h"
 
+
 PGDLLEXPORT Datum _pgr_lengauer_tarjan_dominator_tree (PG_FUNCTION_ARGS);
 PG_FUNCTION_INFO_V1(_pgr_lengauer_tarjan_dominator_tree );
```
Comment (cvvergara) [inline src/LTDTree/LTDTree.c:44]: We use camel case
So your function names should be:
`_pgr_lengauertarjandominatortree` in the C file
and `_pgr_lengauerTarjanDominatorTree`  `pgr_lengauerTarjanDominatorTree` on the SQL files

File: .vscode/settings.json [hunk 8]
```diff
@@ -0,0 +1,66 @@
+{
```
[the remark this answers is not in the document]
Comment (prakashupes) [reply .vscode/settings.json:1]: I'm using vscode , so I just pushed it by mistake.

File: include/bipartite/pgr_bipartite_driver.hpp (hunk 1 above)
Comment (prakashupes) [reply include/bipartite/pgr_bipartite_driver.hpp:65]: okay ma'am

File: include/bipartite/pgr_bipartite_driver.hpp (hunk 2 above)
Comment (prakashupes) [reply include/bipartite/pgr_bipartite_driver.hpp:72]: done

File: include/bipartite/pgr_bipartite_driver.hpp (hunk 3 above)
Comment (prakashupes) [reply include/bipartite/pgr_bipartite_driver.hpp:77]: done

File: include/bipartite/pgr_bipartite_driver.hpp (hunk 4 above)
Comment (prakashupes) [reply include/bipartite/pgr_bipartite_driver.hpp:71]: done

File: src/LTDTree/LTDTree.c (hunk 7 above)
Comment (prakashupes) [reply src/LTDTree/LTDTree.c:44]: I need to fix the name in all pgTap and docs where I used ```_pgr_lengauer_tarjan_dominator_tree```.

**Comment by prakashupes** [comment]
> What happened? why Prim & Kruskal are messed up?
> https://github.com/pgRouting/GSoC-pgRouting/pull/112/checks?check_run_id=912238179#step:7:225

@cvvergara When I was developing two graphs common spanning tree then I put it inside the ```spanningTree```  folder. I edited the config file for spanningTree, maybe this is the reason for the errors but now I think it should not come. Because I moved all the files of spanningTree in the new branch.

File: src/LTDTree/LTDTree.c (hunk 7 above)
Comment (cvvergara) [reply src/LTDTree/LTDTree.c]: This needs to be done

Comment (prakashupes) [reply src/LTDTree/LTDTree.c]: done ma'am

**Pull request merged**